File: Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression_working.py
# Polynomial Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, [1]].values
y = dataset.iloc[:, 2].values

"""# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting the Model to Linear Regression
from sklearn.linear_model import LinearRegression
linear_regressor = LinearRegression()
linear_regressor.fit(X,y)

#print('If Linear is {}'.format(linear_regressor.predict(6.5)))

# Fitting the Model to Polynomial Regression
from sklearn.preprocessing import PolynomialFeatures
polynomial_features = PolynomialFeatures(degree = 4)
X_poly = polynomial_features.fit_transform(X)
linear_regressor2 = LinearRegression()
linear_regressor2.fit(X_poly,y)
#print('If Polynomial Linear is {}'.format(linear_regressor2.predict(6.5)))


# Visualise the result of Linear Regression
X_grid = np.arange(min(X),max(X), step =0.1)
X_grid=X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,marker ='X',color='red')
plt.plot(X,linear_regressor.predict(X),color='green')
plt.plot(X_grid,linear_regressor2.predict(polynomial_features.fit_transform(X_grid)),color='blue')
plt.title('Poly Linear Regressor')
plt.xlabel('Years')
plt.ylabel('Salary')
plt.show

# Predict the value using Polynomial Linear Regression
linear_regressor.predict(6.5)

# Predict the value using Polynomial Linear Regression
linear_regressor2.predict(polynomial_features.fit_transform(6.5))

File: Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression_working.R
# Polynomial Linear Regression

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$Level, SplitRatio = 0.8)
#training_set = subset(dataset, split == TRUE)
#test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

# Fitting to Linear Regression
linear_regression = lm(formula = Salary ~ . ,data = dataset)

# Fitting into Polynomial Linear Regression
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
polynomial_regression = lm(formula = Salary ~ . ,data = dataset)


# Visualise the Linear Regression
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(linear_regression, newdata = dataset)),
             colour = 'blue') +
  geom_line(aes(x = dataset$Level, y = predict(polynomial_regression, newdata = dataset)),
            colour = 'green') +
  ggtitle('Linear Regression Prediction') +
  xlab('Years') +
  ylab('Salary')


# Predict the values using Linear Regression
y_pred = predict(linear_regression, data.frame(Level = 6.5))


# Predict the values using Polynomial Regression
y_pred = predict(polynomial_regression, data.frame(Level = 6.5,
                                                   Level2 = 6.5^2,
                                                   Level3 = 6.5^3,
                                                   Level4 = 6.5^4))